lib/type-hinting.c

```diff
@@ -145,9 +145,9 @@ type_cast_to_double(const gchar *value, gdouble *out, GError **error)
 
   errno = 0;
   *out = strtod(value, &endptr);
-  if (errno == ERANGE && (*out == HUGE_VAL || *out == -HUGE_VAL))
+  if (errno == ERANGE && (*out >= HUGE_VAL || *out <= -HUGE_VAL))
     success = FALSE;
-  if (*out == 0 && endptr == value)
+  if (*out >= 0 && endptr == value)
```

Comment:
The `strtod` could return with `0` for two reason, of which only one of them is when the `value` is `0`. The other reason is handled right here. (See https://linux.die.net/man/3/strtod) Checking if the `0` is larger or equal to zero does not make sense here.

Reply:
ok. I read the manual and got your point. We can do the following,
```
if ((int)(*out) == 0 && endptr == value)
```

Reply:
Yes.